File: backend/src/models/GeofenceZone.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const GeofenceZone = sequelize.define('GeofenceZone', {
    zone_id: {
      type: DataTypes.UUID,
      defaultValue: DataTypes.UUIDV4,
      primaryKey: true
    },
    nom_zone: {
      type: DataTypes.STRING,
      allowNull: false
    },
    coordonnees_polygone: {
      type: DataTypes.JSONB,
      allowNull: false
    },
    secteur: {
      type: DataTypes.STRING,
      allowNull: false
    },
    territoire: {
      type: DataTypes.STRING,
      allowNull: false
    },
    rayon_buffer: {
      type: DataTypes.DECIMAL(10, 2),
      defaultValue: 300.0
    },
    actif: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    }
  }, {
    tableName: 'geofence_zones',
    indexes: [
      {
        fields: ['secteur']
      },
      {
        fields: ['territoire']
      }
    ]
  });

  return GeofenceZone;
};

File: lib/domain/entities/geofence_zone.dart
import 'package:equatable/equatable.dart';

class GeofenceZone extends Equatable {
  final String zoneId;
  final String nomZone;
  final Map<String, dynamic> coordonneesPolygone; // GeoJSON
  final List<String> merchandiserIds;
  final String secteur;
  final String territoire;
  final double rayonBuffer;

  const GeofenceZone({
    required this.zoneId,
    required this.nomZone,
    required this.coordonneesPolygone,
    required this.merchandiserIds,
    required this.secteur,
    required this.territoire,
    required this.rayonBuffer,
  });

  @override
  List<Object> get props => [
    zoneId,
    nomZone,
    coordonneesPolygone,
    merchandiserIds,
    secteur,
    territoire,
    rayonBuffer,
  ];
}